docs/ImportONNXDefs.md

```diff
@@ -1,46 +1,89 @@
 <!--- SPDX-License-Identifier: Apache-2.0 -->
+# Table of Contents
+  1. [Overview](#overview)
+  2. [Add an Operation](#add_operation)
+  3. [Customize an Operation](#customize)
+  4. [Build](#build)
+  5. [Details about version](#version)
+# Overview <a name="overview"></a>
+Onnx-mlir defines an onnx dialect to represent operations specified by onnx.The onnx dialect is createdd with MLIR table
+gen tool. The definition of each operation is transferred from onnx automatically with a 
+ python script, 
+[utils/gen_onnx_mlir.py](../utils/gen_onnx_mlir.py). 
+This script retrieves operation definition from
+ onnx package to generate ONNXOps.td.inc for dialect table gen and OpBuilderTable.inc for 
+onnx model importer in onnx-mlir. 
+The following sections will describe how to use gen_onnx_mlir.py to add an operation into onnx
+ dialect in onnx-mlir and how to refine the definition of the operation.
 
-# Import ONNX specifications into ONNX-MLIR
+# Add an Operation <a name="add_operation"></a>
+To generate an operation for onnx dialect, add this operation into the dictionary, 
+'version_dict', in gen_onnx_mlir.py. 
+The key of this directory is the operation name and the value is the list of 
+opset for this operation. Usually only the top version opset of this operation (in onnx-mlir/third_party/onnx) is supported. Details about versioning can be found in [version section](#Operation Version).
+With this entry, the script will generate the operation defintion for onnx dialect.
 
-ONNX specifications are defined under `onnx/defs` directory in the ONNX project repository. 
-There is a python script [utils/gen_onnx_mlir.py](../utils/gen_onnx_mlir.py) that automatically generates documents about operations in ONNX (docs/Operations.md). 
-ONNX-MLIR modified this script to import ONNX specifications into ONNX-MLIR. 
-There are two files generated for ONNX MLIR with the modified gen_onnx_mlir.py:
+# Customization <a name="customize"></a>
 
-1. `src/Dialect/ONNX/ONNXOps.td.inc`: Operation definition for MLIR TableGen. `src/Dialect/ONNX/ONNXOps.td` includes this file.
-2. `src/Builder/OpBuildTable.inc`: C++ code for ONNX-MLIR frontend to import operation nodes from ONNX model. `src/Builder/FrontendDialectTransformer.cpp` includes this file.
+## Add Interface and Trait
+By default, all operation has shape inference interface and `NoSideEffect` trait.
+If an opration has `ResultTypeInferenceOpInterface`, use dictionary `OpsWithResultTypeInference`. 
+This interface inferes the type of result tensor, not shape. 
+If an operation has subgraph, it will has interface `HasOnnxSubgraphOpInterface`. 
```

Comment:
I would add a sentence on what to do for a new operation to use one or another interface. It would be helpful if each time we are asking the user to do something, we are specific in which file (and maybe how to grep for the right variable) the changes should go into.

Reply:
@chentong319 It looks like a merge was not completed as I see 
```
HEAD
=====
```
in the md file